### ISSUE TITLE
Trailing closing parentheses in URLs are discarded in href

### ISSUE THREAD

**Issue by ahamez** [pragdave/earmark#342]
I spotted it on https://hexdocs.pm/mix/master/Mix.Tasks.Escript.Install.html: the link to the PATH variable on Wikipedia is `https://en.wikipedia.org/wiki/PATH_(variable`, without the ending `)`, whereas it should be `https://en.wikipedia.org/wiki/PATH_(variable)`, with the ending `)`.
At first I thought the bug originated in ex_docs, but after some investigation, I came to the conclusion that the bug lies in earmak:

```elixir
iex> Earmark.as_ast("https://https://en.wikipedia.org/wiki/PATH_(variable)")
{:ok,
 [
   {"p", [],
    [
      {"a", [{"href", "https://https://en.wikipedia.org/wiki/PATH_(variable"}],
       ["https://https://en.wikipedia.org/wiki/PATH_(variable"]},
      ")"
    ]}
 ], []}

iex> Earmark.as_html("https://https://en.wikipedia.org/wiki/PATH_(variable)")
{:ok,
 "<p><a href=\"https://https://en.wikipedia.org/wiki/PATH_(variable\">https://https://en.wikipedia.org/wiki/PATH_(variable</a>)</p>\n",
 []}
```

Note that all trailing `)` are discarded, no matter how many parentheses.

**Comment by RobertDober** [comment]
Thank you for reporting this, looks bad, as I do not see an easy workaround before I fix this.

Well you could put the link URL encoded that should work, but obviously that is unacceptable, even mid term.

I'll try to squeeze this fix into 1.4.5...

**Comment by RobertDober** [comment]
I mean this is really bad, nothing to do with us being neighbours ;)
